File: TimeKeeper/Models/Workday.cs
using Microsoft.AspNetCore.Routing.Constraints;

namespace TimeKeeper.Models {
    public class Workday {

        public int id { get;set; }
        public int userId { get;set; }
        public DateTime date { get;set; }
        public int? projectId { get;set; }
        public DateTime clockIn { get;set; }
        public DateTime? clockOut { get;set; }
        public double? workHours { get;set; }
        public string? description { get;set; }
        public double? grade { get;set; }
        public string? attachment { get;set; }

        public Workday() { }

        public Workday (int id, int userId, DateTime date, int? projectId, DateTime clockIn, DateTime? clockOut, double? workHours, string? description, double? grade, string? attachment) {
            this.id = id;
            this.userId = userId;
            this.date = date;
            this.projectId = projectId;
            this.clockIn = clockIn;
            this.clockOut = clockOut;
            this.workHours = workHours;
            this.description = description;
            this.grade = grade;
            this.attachment = attachment;
        }

        public Workday(int userId, DateTime date, int? projectId, DateTime clockIn, DateTime? clockOut, double? workHours, string? description, double? grade, string? attachment) {
            this.userId = userId;
            this.date = date;
            this.projectId = projectId;
            this.clockIn = clockIn;
            this.clockOut = clockOut;
            this.workHours = workHours;
            this.description = description;
            this.grade = grade;
            this.attachment = attachment;
        }

        public override bool Equals(object obj) {
            var item = obj as Workday;

            if (item == null) {
                return false;
            }

            if(item.date.Year == this.date.Year && item.date.Month == this.date.Month && item.date.Day == this.date.Day)
            {
                return true;
            }
            else
            {
                return false;
            }

            
        }


    }
}

File: TimeKeeper/ClientApp/src/app/Shared/Models/Workday.ts
export class Workday {

    id: number;
    userId: number;
    date: string;  // e.g. 2023-01-10T00:00:00
    projectId: number;
    clockIn: string;  // e.g. 2023-01-10T00:00:00
    clockOut: string;  // e.g. 2023-01-10T00:00:00
    workHours: number;
    description: string;
    grade: number;
    attachment: string | null;

    constructor(workday_obj: any) {
        this.id = workday_obj["id"];
        this.userId = workday_obj["userId"];
        this.date = workday_obj["date"];
        this.projectId = workday_obj["projectId"];
        this.clockIn = workday_obj["clockIn"];
        this.clockOut = workday_obj["clockOut"];
        this.workHours = workday_obj["workHours"];
        this.description = workday_obj["description"];
        this.grade = workday_obj["grade"];
        this.attachment = workday_obj["attachment"];
    }

    public getClockInTime(): string {
        return this.clockIn.split("T")[1];
    }

    public getClockOutTime(): string {
        return this.clockOut.split("T")[1];
    }

    public getDateIso(): string {
        return this.date.split("T")[0];
    }
}
